actors/builtin/miner/expirations.go

```diff
@@ -0,0 +1,29 @@
+package miner
+
+import (
+	"sort"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+)
+
+type sectorEpochSet struct {
+	epoch   abi.ChainEpoch
+	sectors []uint64
+}
+
+func groupSectorsByExpiration(sectors []*SectorOnChainInfo) []sectorEpochSet {
+	sectorsByExpiration := make(map[abi.ChainEpoch][]uint64)
+
+	for _, sector := range sectors {
+		sectorsByExpiration[sector.Expiration] = append(sectorsByExpiration[sector.Expiration], uint64(sector.SectorNumber))
+	}
+
+	sectorEpochSets := make([]sectorEpochSet, 0, len(sectorsByExpiration))
+	for expiration, sectors := range sectorsByExpiration {
```

Comment:
The map iteration makes me nervous, though I think this one's ok since no side-effects. We should probably document every map iteration with "this one's ok because it has no side effects or runtime interactions" to help avoid someone changing something that breaks that.

Reply:
I've added some documentation.